### ISSUE TITLE
Segmentation fault with additive constant

### ISSUE THREAD

**Issue by rlaboiss** [praat/praat#1782]
Praat v.6.1.48 is segfaulting on my Debian unstable amd64 system:

```
$ ./praat dwtest/test_MDS.praat 
test_MDS.praat
        test interface
                Query
Segmentation fault
```

I narrowed down the problem, resulting in this minimal script that exposes the bug on my system:

```
.dissimilarity = Create letter R example: 0
.additiveConstant = Get additive constant
```

Here is the strace log, if it can be of any help: [additive-constant-failure.txt](https://github.com/praat/praat/files/6616264/additive-constant-failure.txt)

**Comment by rlaboiss** [comment]
It seems that the problem is tied to the “letter R” parameter. The following script:

```
.dissimilarity = Create letter R example: 1
.additiveConstant = Get additive constant
```

produces the following

```
$  ./praat additive-constant-failure.praat
Error: There are no positive dissimilarities.
Additive constant not calculated.
Script line 2 not performed or completed:
« .additiveConstant = Get additive constant »
Script “/var/tmp/additive-constant-failure.praat” not completed.
Praat: script command <</var/tmp/additive-constant-failure.praat>> not completed.
```

**Comment by davidweenink** [comment]
I have included a better test such that the exception will not occur anymore (hopefully).
However, I cannot find any reason why your test should segfault because Praat gives a message that something is wrong.

**Comment by rlaboiss** [comment]
> I have included a better test such that the exception will not occur anymore (hopefully).

Thanks. Could you please tell me which commit fixes it? The version of `dwtest/test_MDS.praat` in HEAD (commit a764c4bf78371d1d19b702e287126d8bb33822ae) still segfaults.

> However, I cannot find any reason why your test should segfault because Praat gives a message that something is wrong.

It segfaults only when the argument of the command `Create letter R example:` is 0. In this case, Praat does not give any message (see my [original report](https://github.com/praat/praat/issues/1782#issue-914936544)).

**Comment by davidweenink** [comment]
Dear Rafael,

it turned out that things went wrong because the use of initialisation lists for arrays turned out to have some unexpected behaviour. There is not one commit that fixes it but in any case since my latest commit dd0d78d37afdd0669d0ba035245ea7c3a5b169a8 on github the dust has settled down.

Greetings,

David

________________________________

**Comment by rlaboiss** [comment]
It is still failing with Praat 6.1.49:

```
$ ./praat --version
Praat 6.1.49 (June 12 2021)
$ (cd dwtest ; ../praat --run runAllTests_batch.praat)
[…]
### executing test_MDS.praat:
test_MDS.praat
        test interface
                Query
                Modify: skipped
                Synthesize: skipped
                Extract part
                Extract
                Convert
                To Configuration
Current 1: minimum = 9.362233557752591e-18
                Dissimilarity & Configuration
Current 1: minimum = 9.282136142788268e-18
Error: Script assertion fails in line 215 (false):
   .stress1 <= .stress1_random; --undefined-- <= 0.42405501019230496 ? 1 conf[1]
```

**Comment by rlaboiss** [comment]
It is still failing with Praat 6.1.49:

```
$ ./praat --version
Praat 6.1.49 (June 12 2021)
$ (cd dwtest ; ../praat --run runAllTests_batch.praat)
[…]
### executing test_MDS.praat:
test_MDS.praat
        test interface
                Query
                Modify: skipped
                Synthesize: skipped
                Extract part
                Extract
                Convert
                To Configuration
Current 1: minimum = 9.362233557752591e-18
                Dissimilarity & Configuration
Current 1: minimum = 9.282136142788268e-18
Error: Script assertion fails in line 215 (false):
   .stress1 <= .stress1_random; --undefined-- <= 0.42405501019230496 ? 1 conf[1]
```

**Comment by davidweenink** [comment]
I restructured the MDS testing. Hopefully the new test_MDS.praat survives your testing too.

**Comment by rlaboiss** [comment]
> I restructured the MDS testing. Hopefully the new test_MDS.praat survives your testing too.

Thanks, @davidweenink, but it is still failing for me:

```
$ (cd dwtest ; ../praat --run test_MDS.praat)
test_MDS.praat
        test Dissimilarity letter R
Error: Script assertion fails in line 40 (false):
   object[.dissimilarity, 1, 2] = 6
```

For my build of Praat (v6.1.49), all off-diagonal elements of the dissimilarity matrix for the “Lettter R” example have value equal 5, hence the failing assertion above.

**Comment by davidweenink** [comment]
this could mean that the rest of your praat code is not uptodate yet or that the test is not good enough on you platform. Because it tests the equality of two floating point numbers which is potentially dangerous. 
I think I have to make a better test.

**Comment by rlaboiss** [comment]
> this could mean that the rest of your praat code is not uptodate yet or that the test is not good enough on you platform. Because it tests the equality of two floating point numbers which is potentially dangerous.

Indeed, I had an old version of file `dwtools/Configuration.cpp`. Everything is working now, and I release a patched version 6.1.49 of the Debian package for Praat.

Thanks!
